plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/p5edges/orthogonal/OrthogonalRoutingGenerator.java

```diff
@@ -188,6 +232,44 @@ public int routeEdges(final IElkProgressMonitor monitor, final LGraph layeredGra
         return rankCount + 1;
     }
 
+    /**
+     * Computes and returns the minimum distance between any two adjacent source connections and any two adjacent target
+     * connections.
+     */
+    private double minimumHorizontalSegmentDistance(final List<HyperEdgeSegment> edgeSegments) {
+        double minIncomingDistance = minimumDifference(
+                edgeSegments.stream().flatMap(segment -> segment.getIncomingConnectionCoordinates().stream()));
+        double minOutgoingDistance = minimumDifference(
+                edgeSegments.stream().flatMap(segment -> segment.getOutgoingConnectionCoordinates().stream()));
+        
+        return Math.min(minIncomingDistance, minOutgoingDistance);
+    }
+    
+    /**
+     * Returns the smallest difference between any two numbers in the given stream of numbers. If there are less than
+     * two numbers, we return {@link Double#MAX_VALUE}.
+     */
+    private double minimumDifference(final Stream<Double> numberStream) {
+        List<Double> numbers = numberStream.sorted().distinct().collect(Collectors.toList());
+        
+        double minDifference = Double.MAX_VALUE;
+        
+        if (numbers.size() >= 2) {
+            Iterator<Double> iter = numbers.iterator();
```

Comment:
Maybe implement using a `Stream::reduce(...)`?

Reply:
I don't see how. Here, we always consider two consecutive elements to determine their difference. The `reduce(...)` operations I know only ever consider a single element at a time. Am I missing something?

Reply:
Maybe along these lines: 
```
        Stream<Pair<Double, Double>> subsequentPairs = Streams.zip(s, s.skip(1), (a, b) -> Pair.of(a, b));
        double minDiff = subsequentPairs.reduce(Double.POSITIVE_INFINITY,
                (min, pair) -> pair.getValue() - pair.getKey(), 
                Math::min);
```

Reply:
But isn't the point of using streams making things _more_ understandable? This I feel I have to decipher first. 😄

Reply:
Yes, and less error-prone by stating _what_ is done, not _how_. 

With streams you have basically two tasks (that could be explained using a short comment):
1. create a list of pairs of subsequent elements
2. find the minimum of the differences between the pairs' elements

The rest is getting used to the (partly awkward) syntax of java :).